File: frontend/lib/models/book_review.dart
class BookReview {
  BookReview({
    required this.id,
    required this.bookId,
    required this.userId,
    required this.rating,
    this.title,
    this.comment,
    this.user,
    this.createdAt,
    this.updatedAt,
    // NEW: Vote counts and user vote
    this.helpfulCount = 0,
    this.notHelpfulCount = 0,
    this.userVote,
  });

  final int id;
  final int bookId;
  final int userId;
  final int rating;
  final String? title;
  final String? comment;
  final Map<String, dynamic>? user;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  // NEW: Vote fields
  final int helpfulCount;
  final int notHelpfulCount;
  final String? userVote; // 'like', 'dislike', or null

  factory BookReview.fromJson(Map<String, dynamic> json) {
    return BookReview(
      id: json['id'] as int,
      bookId: json['book_id'] as int,
      userId: json['user_id'] as int,
      rating: json['rating'] as int,
      title: json['title'] as String?,
      comment: json['comment'] as String?,
      user: json['user'] as Map<String, dynamic>?,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'] as String)
          : null,
      // NEW: Parse vote data
      helpfulCount: json['helpful_count'] as int? ?? 0,
      notHelpfulCount: json['not_helpful_count'] as int? ?? 0,
      userVote: json['user_vote'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'book_id': bookId,
      'user_id': userId,
      'rating': rating,
      'title': title,
      'comment': comment,
      'user': user,
      'created_at': createdAt?.toIso8601String(),
      'updated_at': updatedAt?.toIso8601String(),
      'helpful_count': helpfulCount,
      'not_helpful_count': notHelpfulCount,
      'user_vote': userVote,
    };
  }

  BookReview copyWith({
    int? id,
    int? bookId,
    int? userId,
    int? rating,
    String? title,
    String? comment,
    Map<String, dynamic>? user,
    DateTime? createdAt,
    DateTime? updatedAt,
    int? helpfulCount,
    int? notHelpfulCount,
    String? userVote,
  }) {
    return BookReview(
      id: id ?? this.id,
      bookId: bookId ?? this.bookId,
      userId: userId ?? this.userId,
      rating: rating ?? this.rating,
      title: title ?? this.title,
      comment: comment ?? this.comment,
      user: user ?? this.user,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      helpfulCount: helpfulCount ?? this.helpfulCount,
      notHelpfulCount: notHelpfulCount ?? this.notHelpfulCount,
      userVote: userVote ?? this.userVote,
    );
  }
}

File: backend/app/Models/BookReview.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class BookReview extends Model
{
    use HasFactory;

    protected $fillable = [
        'book_id',
        'user_id',
        'rating',
        'title',
        'comment',
        'helpful_count',
        'not_helpful_count',
    ];

    protected $casts = [
        'rating' => 'integer',
        'helpful_count' => 'integer',
        'not_helpful_count' => 'integer',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    protected $appends = ['user_vote'];

    /**
     * Get the book that owns the review
     */
    public function book(): BelongsTo
    {
        return $this->belongsTo(Book::class);
    }

    /**
     * Get the user that owns the review
     */
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Get the votes for the review
     */
    public function votes(): HasMany
    {
        return $this->hasMany(ReviewVote::class, 'review_id');
    }

    /**
     * Get current authenticated user's vote
     */
    public function getUserVoteAttribute(): ?string
    {
        if (!auth()->check()) {
            return null;
        }

        $vote = $this->votes()
            ->where('user_id', auth()->id())
            ->first();

        return $vote ? $vote->vote_type : null;
    }

    /**
     * Update vote counts from database
     */
    public function updateVoteCounts(): void
    {
        $this->helpful_count = $this->votes()->where('vote_type', 'like')->count();
        $this->not_helpful_count = $this->votes()->where('vote_type', 'dislike')->count();
        $this->saveQuietly(); // Save without triggering events
    }

    /**
     * Scope to include vote counts
     */
    public function scopeWithVoteCounts($query)
    {
        return $query->withCount([
            'votes as helpful_count' => function ($query) {
                $query->where('vote_type', 'like');
            },
            'votes as not_helpful_count' => function ($query) {
                $query->where('vote_type', 'dislike');
            },
        ]);
    }

    /**
     * Transform for API response
     */
    public function toArray()
    {
        $array = parent::toArray();
        
        // Include user info
        if ($this->relationLoaded('user')) {
            $array['user'] = [
                'id' => $this->user->id,
                'name' => $this->user->name,
                'email' => $this->user->email,
            ];
        }

        return $array;
    }
}
